dd-sdk-android/src/test/kotlin/com/datadog/android/error/internal/DatadogExceptionHandlerTest.kt

```diff
@@ -0,0 +1,167 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-2020 Datadog, Inc.
+ */
+
+package com.datadog.android.error.internal
+
+import android.util.Log as AndroidLog
+import com.datadog.android.Datadog
+import com.datadog.android.core.internal.data.Writer
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.log.assertj.LogAssert.Companion.assertThat
+import com.datadog.android.log.forge.Configurator
+import com.datadog.android.log.internal.domain.Log
+import com.datadog.android.log.internal.net.NetworkInfo
+import com.datadog.android.log.internal.net.NetworkInfoProvider
+import com.datadog.android.log.internal.user.UserInfo
+import com.datadog.android.log.internal.user.UserInfoProvider
+import com.datadog.android.utils.mockContext
+import com.datadog.tools.unit.extensions.ApiLevelExtension
+import com.datadog.tools.unit.invokeMethod
+import com.nhaarman.mockitokotlin2.argumentCaptor
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.verifyZeroInteractions
+import com.nhaarman.mockitokotlin2.whenever
+import fr.xgouchet.elmyr.Forge
+import fr.xgouchet.elmyr.annotation.Forgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import java.util.Date
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class),
+    ExtendWith(ApiLevelExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DatadogExceptionHandlerTest {
+
+    lateinit var testedHandler: DatadogExceptionHandler
+
+    @Mock
+    lateinit var mockPreviousHandler: Thread.UncaughtExceptionHandler
+    @Mock
+    lateinit var mockNetworkInfoProvider: NetworkInfoProvider
+    @Mock
+    lateinit var mockTimeProvider: TimeProvider
+    @Mock
+    lateinit var mockUserInfoProvider: UserInfoProvider
+    @Mock
+    lateinit var mockLogWriter: Writer<Log>
+
+    @Forgery
+    lateinit var fakeThrowable: Throwable
+    @Forgery
+    lateinit var fakeNetworkInfo: NetworkInfo
+    @Forgery
+    lateinit var fakeUserInfo: UserInfo
+    @Forgery
+    lateinit var fakeTime: Date
+
+    @BeforeEach
+    fun `set up`(forge: Forge) {
```

Comment:
I think you should have a tearDown method where you restore the original handler otherwise you will have issues in your tests as that field is static and it the process will keep the latest value you allocated from your test method.

Reply:
Good call !